src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -397,7 +402,22 @@ private void tearDownJabRef(List<String> filenames) {
      * @return true if the user chose to quit; false otherwise
      */
     public boolean quit() {
-        // First ask if the user really wants to close, if the library has not been saved since last save.
+        // First ask if the user really wants to close, if there are still background tasks running
+        /*
+        It is important to wait for unfinished background tasks before checking if a save-operation is needed, because
+        the background tasks may make changes themselves that need saving.
+         */
+        if (stateManager.getAnyTaskRunning().getValue()) {
+            if (!(dialogService.showBackgroundProgressDialogAndWait(
+                    Localization.lang("Please wait..."),
+                    Localization.lang("Waiting for background tasks to finish. Quit anyway?"),
+                    stateManager
+            ).orElse(ButtonType.CANCEL) == ButtonType.YES)) {
```

Comment:
This looks odd to me

Reply:
Could you please elaborate?
If you mean the condition at the end, I just want to make sure that JabRef only exists if the user pressed yes. So if the optional is empty I just put ButtonType.CANCEL so the comparison to YES returns false.
Do you want me to change this or add a comment? Maybe it would be clearer to store the result in a variable and check for (result.isPresent() && result.get() == ButtonType.YES).

Reply:
Yes, I think this is more readable.

Reply:
Done https://github.com/JabRef/jabref/pull/6443/commits/3db399741b0ce098cd930005870bc745cd719c70